src/cloudWatchLogs/utils.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { CLOUDWATCH_LOGS_SCHEME } from './constants'
+
+// URIs are the only vehicle for delivering information to a TextDocumentContentProvider.
+// The following functions are used to structure and destructure relevant information to/from a URI.
+// Colons are not valid characters in either the group name or stream name and will be used as separators.
+
+/**
+ * Destructures an awsCloudWatchLogs URI into its component pieces.
+ * @param uri URI for a Cloudwatch Logs file
+ */
+export function convertUriToLogGroupInfo(
```

Comment:
typically this is called "parse", but I see the duality with `convertLogGroupInfoToUri` ...